data/transactions/logic/eval.go

```diff
@@ -331,15 +387,14 @@ func NewEvalParams(txgroup []transactions.SignedTxnWithAD, proto *config.Consens
 		FeeCredit:               &credit,
 		PooledApplicationBudget: pooledApplicationBudget,
 		pooledAllowedInners:     pooledAllowedInners,
-		created:                 &resources{},
+		available:               &resources{boxes: allBoxes},
 		appAddrCache:            make(map[basics.AppIndex]basics.Address),
```

Comment:
@jannotti Sanity checking my understanding:  Are `ioBudget` and `readBudgetChecked` intentionally _not_ included in `NewEvalParams`?

Based on my understanding, I thought these parameters will be initialized to non-default values here to pass state through to inner-transaction processing.

Reply:
readBudgetChecked will get false, because that's what Go does. And because it gets false, ioBudget will get initialized in the first EvalContract done using the EvalParams.

Reply:
Resolving - Touched base with @jannotti verbally and the discussion addressed my concerns.

As a clarification for completeness:   I left the original comment in the wrong place.  I intended to discuss `NewInnerEvalParams`.  _Not_ NewEvalParams`.